src/formatDate.js

```diff
@@ -8,7 +8,84 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatObject = arrayInObject(fromFormat);
+  const toFormatObject = arrayInObject(toFormat);
+  const arrayOldDate = date.split(fromFormat[3]);
+  const dateObject = arrayDateFromObject(arrayOldDate, fromFormat);
+
+  if (fromFormatObject.yy !== toFormatObject.yy) {
+    if (fromFormatObject.yy.length === 4) {
+      dateObject.yy = dateObject.yy.slice(2);
+    } else {
+      if (+dateObject.yy < 30) {
```

Comment:
It's good that you're considering the century when converting from `YY` to `YYYY`. However, ensure that `dateObject.yy` is a string and that the conversion is necessary (i.e., `fromFormatObject.yy.length` is 2).